spec/acceptance/should_create_task_spec.rb

```diff
@@ -67,4 +75,190 @@
+        start_time => '12:00',
+      },
+      provider      => 'win32_taskscheduler'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+
+    # Get a new username and password
+    username_2, password_2 = add_test_user(host)
+
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      compatibility => 1,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{username_2}',
+      password      => '#{password_2}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'win32_taskscheduler'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+
+    remove_test_user(host, username_2)
+  end
+
+  it "Should update a task's credentials" do
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      compatibility => 1,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{@username}',
+      password      => '#{@password}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'taskscheduler_api2'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+
+    # Get a new username and password
+    username_2, password_2 = add_test_user(host)
+
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      compatibility => 1,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{username_2}',
+      password      => '#{password_2}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'taskscheduler_api2'
```

Comment:
Its not necessary to specify this provider - its the default. You can also omit `compatiiblity` which defaults to `1`